cloudinit/distros/freebsd.py

```diff
@@ -192,5 +192,29 @@ def update_package_sources(self):
             freq=PER_INSTANCE,
         )
 
-
-# vi: ts=4 expandtab
+    def is_container(self) -> bool:
+        jailed, _ = subp.subp(["sysctl", "-n", "security.jail.jailed"])
+        if jailed.strip() == "0":
+            return False
+        return True
+
+    def virtual(self) -> str:
+        if self.is_container():
+            return "jail"
+        # map FreeBSD's kern.vm_guest to systemd-detect-virt, just like we do
+        # in ds-identify
+        VM_GUEST_TO_SYSTEMD = {
```

Comment:
no bhyve or not this place?

Reply:
'bhyve' is reported by both `kern.vm_guest` and `systemd-detect-virt`, so no *translation* is necessary.

that's what that table is for